### DISCUSSION TITLE
Restore windows position on external monitor

### DISCUSSION THREAD

**Discussion by kamzata** [rxhanson/Rectangle#944] [General]
I'm using a Macbook Air 13" connected to an external monitor (5120x1440px). Whenever I disconnect the monitor and then reconnect it again all the windows are moved and compressed on top left of my external monitor. I'd like to save the windows position before disconnect the monitor and then restore them once reconnected. Or better... once the computer recognizes that is the same monitor which was connected before it should restore them automatically.

Does anyone have the same problem? That's a really annoying for me.

**Comment by rxhanson** [comment]
As far as I know, there's not a way to know if macOS is about to reflect changes from a monitor being disconnected, you can only get a notification that a monitor has been disconnected, so you wouldn't be able to quickly save off window positions, unless you were constantly polling prior to a disconnect. Alternatively, you could leverage history of where things are placed by Rectangle, which of course wouldn't recognize manually moved windows. I don't have the bandwidth to do something like that at this time, but would welcome a good pull request for it.

As a paid alternative, you could save a layout in Rectangle Pro, and have it automatically perform that layout when a display is connected and/or disconnected.

**Reply by kamzata** [reply]
> As far as I know, there's not a way to know if macOS is about to reflect changes from a monitor being disconnected, you can only get a notification that a monitor has been disconnected, so you wouldn't be able to quickly save off window positions, unless you were constantly polling prior to a disconnect. Alternatively, you could leverage history of where things are placed by Rectangle, which of course wouldn't recognize manually moved windows. I don't have the bandwidth to do something like that at this time, but would welcome a good pull request for it.
> 
> As a paid alternative, you could save a layout in Rectangle Pro, and have it automatically perform that layout when a display is connected and/or disconnected.

Maybe you can intercept an event anytime a window is moved and save it... however, I'll try Rectangle Pro. Thanks

**Reply by bigplayer-ai** [reply]
Feature Request: Remember Window Positions When Disconnecting from Monitors
## Retain Window Positions on Monitor Disconnect

Hello,

I'm an avid user of **Rectangle** and find it incredibly useful for managing windows on my Mac. However, I've observed that the window positions aren't retained when I disconnect my laptop from external monitors, requiring me to reorganize them upon every reconnection.

In 2023, I'm curious to know if **Rectangle Pro** offers a feature to remember window positions even when monitors are disconnected?
